app/controllers/application_controller.rb

```diff
@@ -25,8 +25,8 @@ def append_info_to_payload(payload)
     payload[:ip] = request.remote_ip
   end
 
-  def analytics
-    @analytics ||= Analytics.new(current_user, request_attributes, ahoy)
+  def analytics(service = Analytics.new(current_user, request))
+    @analytics ||= service
```

Comment:
What I mean is that `analytics` can be called many times in the same request with different args, but it will only honor the arg on the first call, right?